lib/compiler/node/wasm_exec.ts

```diff
@@ -11,33 +11,43 @@ import fs from 'fs';
 import crypto from 'crypto';
 import { TextEncoder, TextDecoder } from 'util';
 
-Object.defineProperties(globalThis, {
-  fs: {
+if (!globalThis.fs) {
+  Object.defineProperty(globalThis, 'fs', {
     value: fs,
-    enumerable: true,
-  },
-  crypto: {
+  });
+}
+
+if (!globalThis.process) {
+  Object.defineProperties(globalThis, 'process', {
+    value: process,
+  });
+}
+
+if (!globalThis.crypto) {
+  Object.defineProperty(globalThis, 'crypto', {
     value: {
       getRandomValues(b) {
         crypto.randomFillSync(b);
       },
     },
-    enumerable: true,
-  },
-  performance: {
+  });
+}
+
+if (!globalThis.performance) {
+  Object.defineProperty(globalThis, 'performance', {
     value: {
       now() {
         const [sec, nsec] = process.hrtime();
         return sec * 1000 + nsec / 1000000;
       },
     },
-    enumerable: true,
-  },
-});
+  });
+}
+// End of polyfills for common API.
```

Comment:
Ran into some problems where `@astropub/webapi` polyfilling `crypto` would cause this to throw. Switched to verify that patching needs to happen before doing it.